integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -720,7 +723,9 @@ object CarbonDataRDDFactory {
 
       // because partitionId=segmentIdIndex*parallelism+RandomPart and RandomPart<parallelism,
       // so segmentIdIndex=partitionId/parallelism, this has been verified.
+      val conf = new SerializableConfiguration(hadoopConf)
```

Comment:
Please copy only necessary configurations instead of serializing whole object. Better use new Serializable class and copy the useful properties to it.